include/openthread/dns.h

```diff
@@ -53,14 +55,11 @@ extern "C" {
 
 #define OT_DNS_MAX_NAME_SIZE 255 ///< Maximum name string size (includes null char at the end of string).
 
-#define OT_DNS_MAX_LABEL_SIZE 64 ///< Maximum label string size (include null char at the end of string)
+#define OT_DNS_MAX_LABEL_SIZE 64 ///< Maximum label string size (include null char at the end of string).
 
-/**
- * Initializer for otDnsTxtIterator.
- */
-#define OT_DNS_TXT_ITERATOR_INIT 0
+#define OT_DNS_TXT_KEY_MIN_LENGTH 1 ///< Minimum length of TXT record key string (RFC 6763 - section 6.4).
 
-typedef uint16_t otDnsTxtIterator; ///< Used to iterate through the TXT entries.
+#define OT_DNS_TXT_KEY_MAX_LENGTH 9 ///< Maximum length of TXT record key string (RFC 6763 - section 6.4).
```

Comment:
Note that RFC 6763 specifies this as a SHOULD and not a MUST.

Reply:
Good point. Will change this to say: "Recommended maximum length ..."